app/src/components/ButtonComponent/ButtonComponent.tsx

```diff
@@ -0,0 +1,132 @@
+import { StyleSheet, TouchableWithoutFeedback, View } from "react-native";
+import React, { useEffect, useState } from "react";
+import NormalText from "../NormalText/NormalText";
+import { Link } from "../NormalText/FontTypes";
+
+interface ButtonComponentInterface {
+  flex?: number;
+  width?: number;
+  height?: number;
+  backgroundColor?: string;
+  disabledColor?: string;
+  isDisabled?: boolean;
+  buttonText: string;
+  textColor?: string;
+  onPress?: (value: string) => void;
+}
+
+const ButtonComponent = ({
+  flex,
+  width,
+  height,
+  backgroundColor,
+  disabledColor,
+  isDisabled,
+  buttonText,
+  textColor,
+  onPress,
+}: ButtonComponentInterface) => {
+  const [positionTop, setPositionTop] = useState(4);
+  const [positionLeft, setPositionLeft] = useState(4);
+  const [focused, setFocused] = useState(false);
+
+  useEffect(() => {
+    if (focused) {
+      setPositionTop(0);
+      setPositionLeft(0);
+    } else {
+      setPositionTop(4);
+      setPositionLeft(4);
+    }
+  }, [focused]);
```

Comment:
Is there another way to handle the focus state?

Reply:
I removed `useEffect` and directly added the state to `onPressIn` and `onPressOut` function. 
What do you think?